### PR TITLE
docs(concepts) make islands.md examples run without errors

### PR DESCRIPTION
**Pull request by rawlingsr**
Addresses https://github.com/denoland/fresh/issues/1582

I've tried to break up my changes so that only the first one or two commits can be taken if they are the only ones that are helpful.

### CODE REVIEW

File: docs/latest/concepts/islands.md [hunk 1]
```diff
@@ -31,6 +31,60 @@ export default function MyIsland() {
...
+// islands/my-island.tsx
+
+import { useSignal } from "@preact/signals";
+import { VNode } from "preact";
+
+interface Data {
+  children: VNode<Element>;
+}
+
+export default function MyIsland({ children }: Data) {
```
Comment (marvinhagemeister) [inline docs/latest/concepts/islands.md]:
```suggestion
import { ComponentChildren } from "preact";

interface Props {
  children: ComponentChildren;
}

export default function MyIsland({ children }: Props) {
```

File: docs/latest/concepts/islands.md [hunk 2]
```diff
@@ -77,18 +113,32 @@ Islands can be nested within other islands as well. In that scenario they act
...
+interface Data {
+  children: VNode<Element>;
+  foo: string;
+}
+
+function randomNumber() {
+  return Math.floor(Math.random() * 100);
+}
+
+export default function MyIsland({ children, foo }: Data) {
```
Comment (marvinhagemeister) [inline docs/latest/concepts/islands.md]:
```suggestion
import { ComponentChildren } from "preact";

interface Props {
  children: ComponentChildren;
  foo: string;
}

function randomNumber() {
  return Math.floor(Math.random() * 100);
}

export default function MyIsland({ children, foo }: Props) {
```

**Review by marvinhagemeister — COMMENTED** [review]
Thanks for making a PR! This is great!!

Left two minor nit comments because the first argument to components is usually called `Props` not  `Data`. It's more of a convention not a hard rule, but that makes it easier for folks coming from other frameworks.

The correct type for `children` is `ComponentChildren` in Preact instead of `VNode<Element>`

**Comment by rawlingsr** [comment]
@marvinhagemeister Thank you for the changes, and for the feedback. 

I wanted to make one other change, but I am not sure what the text should say instead of what it says now. There is a paragraph that talks about not needing to serialize children: https://github.com/setibeings/fresh/blob/eafe8a901a3c9496561cc6b0059882a44e649747/docs/latest/concepts/islands.md?plain=1#L105C1-L109C1 

It uses the pronoun "we" which I think could trip somebody up if they aren't sure if they're reading about something that THEY should be able to tell by looking at the HTML returned by the server.

**Comment by marvinhagemeister** [comment]
@setibeings Good catch. Looks like that is a copy & paste error from the blog post on my part. That section is meant to refer to Fresh the framework itself.

**Comment by rawlingsr** [comment]
I'm not sure if this is the blog post, but I'll just plan on reading it today and take a crack at rephrasing the paragraph I mentioned. 

https://deno.com/blog/intro-to-islands

**Comment by marvinhagemeister** [comment]
@setibeings it references a section from this blog post https://deno.com/blog/fresh-1.2#passing-jsx-to-islands-and-nesting-islands-within-each-other

**Comment by marvinhagemeister** [comment]
Looks like there is a formatting error. Running `deno fmt` locally will resolve these and make the CI green 👍

**Comment by rawlingsr** [comment]
Converting to draft because I now want to add an HTML Example.

**Comment by rawlingsr** [comment]
Take a look at what I wrote and let me know if it matches what you think when you get a moment. Thanks for bearing with me.

**Review by marvinhagemeister — APPROVED** [review]
LGTM, thanks for making the PR 💯

**Pull request merged**